http-client-tck/src/main/java/io/micronaut/http/client/tck/tests/ContinueTest.java

```diff
@@ -0,0 +1,125 @@
+
+    @ParameterizedTest(name = "blocking={0}")
+    @ValueSource(booleans = {true, false})
+    public void testContinueFull(boolean blocking) throws IOException {
+        asserts(SPEC_NAME,
+            Map.of(BLOCKING_CLIENT_PROPERTY, blocking),
+            HttpRequest.POST("/continue/plain", "1")
+                .header(HttpHeaders.EXPECT, "100-continue"),
+            (server, request) -> AssertionUtils.assertDoesNotThrow(server, request,
+                HttpResponseAssertion.builder()
+                    .status(HttpStatus.OK)
+                    .body(BodyAssertion.builder().body("1").equals())
+                    .build())
+        );
+    }
+
+    @ParameterizedTest(name = "blocking={0}")
+    @ValueSource(booleans = {true, false})
+    public void testContinuePublisher(boolean blocking) throws IOException {
+        asserts(SPEC_NAME,
+            Map.of(BLOCKING_CLIENT_PROPERTY, blocking),
+            HttpRequest.POST("/continue/plain", Flux.just("1", "2"))
+                .header(HttpHeaders.EXPECT, "100-continue"),
+            (server, request) -> AssertionUtils.assertDoesNotThrow(server, request,
+                HttpResponseAssertion.builder()
+                    .status(HttpStatus.OK)
+                    .body(BodyAssertion.builder().body("[1,2]").equals())
+                    .build())
+        );
+    }
+
+    @Tag("multipart")
+    @ParameterizedTest(name = "blocking={0}")
+    @ValueSource(booleans = {true, false})
+    public void testContinueMultipart(boolean blocking) throws IOException {
+        asserts(SPEC_NAME,
+            Map.of(BLOCKING_CLIENT_PROPERTY, blocking),
+            HttpRequest.POST("/continue/part", MultipartBody.builder().addPart("foo", "bar").build())
+                .header(HttpHeaders.EXPECT, "100-continue")
+                .contentType(MediaType.MULTIPART_FORM_DATA),
+            (server, request) -> AssertionUtils.assertDoesNotThrow(server, request,
+                HttpResponseAssertion.builder()
+                    .status(HttpStatus.OK)
+                    .body(BodyAssertion.builder().body("bar").equals())
+                    .build())
+        );
+    }
+
+    @ParameterizedTest(name = "blocking={0}")
+    @ValueSource(booleans = {true, false})
+    public void testContinueFormEncoded(boolean blocking) throws IOException {
+        asserts(SPEC_NAME,
+            Map.of(BLOCKING_CLIENT_PROPERTY, blocking),
+            HttpRequest.POST("/continue/part", Map.of("foo", "bar"))
+                .header(HttpHeaders.EXPECT, "100-continue")
+                .contentType(MediaType.APPLICATION_FORM_URLENCODED),
+            (server, request) -> AssertionUtils.assertDoesNotThrow(server, request,
+                HttpResponseAssertion.builder()
+                    .status(HttpStatus.OK)
+                    .body(BodyAssertion.builder().body("bar").equals())
+                    .build())
+        );
+    }
+
+    @Controller("/continue")
```

Comment:
Please, annotate the controller with `@Requires(property = spec.name` to avoid bean pollution between tests